File: ts/easy/scoreOfString.ts
/*
  You are given a string s. The score of a string is defined as the sum of the absolute difference between the ASCII values of adjacent characters.
  Return the score of s.
*/

/**
 * @param {string} s
 * @return {number}
 */
var scoreOfString = function(s: string): number {
  let result = 0;
  for (let i = 1; i < s.length; i++) {
      const prev = s[i - 1].charCodeAt(0);
      const current = s[i].charCodeAt(0);
      result += Math.abs(prev - current);
  }
  return result;
};

File: python/easy/score_of_string.py
"""
  You are given a string s. The score of a string is defined as the sum of the absolute difference between the ASCII values of adjacent characters.
  Return the score of s.
"""

class Solution:
    def scoreOfString(self, s: str) -> int:
        result = 0
        for i in range(1, len(s)):
            result += abs(int(ord(s[i-1]) - int(ord(s[i]))))
        return result